### ISSUE TITLE
master is broken ?

### ISSUE THREAD

**Issue by lefessan** [OCamlPro/typerex-lint#83]
While compiling master with 4.02.1, I get an error with a cyclic abbreviation in ast_element.ml. Looking at the code generated by the ppx with `-dsource`, I get:

```
type record_field = (Longident.t Asttypes.loc* Parsetree.expression)
open Parsetree
open Asttypes
module Exploded =
  struct
    type attribute = (string loc* payload)
    and extension = (string loc* payload)
    and attributes = attribute list
    and payload = payload =
      | PStr of structure
      | PTyp of core_type
      | PPat of pattern* expression option[@@nonrec ]
    and core_type = core_type =
      {
      ptyp_desc: core_type_desc;
      ptyp_loc: Location.t;
      ptyp_attributes: attributes;}[@@nonrec ]
    and core_type_desc = core_type_desc =
```
and many other such duplications of the types. Any reason for that ?

**Comment by ocamlpro-Hufschmitt** [comment]
Woups, I had forgotten that the `nonrec` keyword wasn't available in 4.02.1.

The duplication is practical for the code generation because I add some
auxiliary types, and it is easier to have everything accessible under the same
path.

Ill investigate how I can avoid this without addinc complexity to the generator
code.

**Comment by lefessan** [comment]
Thanks, it would be nice

**Comment by ocamlpro-Hufschmitt** [comment]
Should be fixed by #85
